images/vault/tests/main.tf

```diff
@@ -17,54 +18,69 @@ data "oci_string" "ref" {
   input    = each.value
 }
 
-resource "random_pet" "suffix" {}
+data "imagetest_inventory" "this" {}
 
-resource "helm_release" "vault" {
-  name             = "vault-${random_pet.suffix.id}"
-  namespace        = "vault-system-${random_pet.suffix.id}"
-  repository       = "https://helm.releases.hashicorp.com"
-  chart            = "vault"
-  create_namespace = true
+resource "imagetest_harness_k3s" "this" {
+  name      = "vault"
+  inventory = data.imagetest_inventory.this
+}
+
+module "helm" {
+  source = "../../../tflib/imagetest/helm"
 
-  values = [
-    jsonencode({
-      injector = {
-        agentImage = {
-          repository = data.oci_string.ref["vault"].registry_repo
-          tag        = data.oci_string.ref["vault"].pseudo_tag
-        }
-        image = {
-          repository = data.oci_string.ref["vault-k8s"].registry_repo
-          tag        = data.oci_string.ref["vault-k8s"].pseudo_tag
-        }
+  name      = "vault"
+  namespace = "vault-system"
+  chart     = "vault"
+  repo      = "https://helm.releases.hashicorp.com"
+
+  values = {
+    injector = {
+      agentImage = {
+        repository = data.oci_string.ref["vault"].registry_repo
+        tag        = data.oci_string.ref["vault"].pseudo_tag
       }
-      server = {
-        image = {
-          repository = data.oci_string.ref["vault"].registry_repo
-          tag        = data.oci_string.ref["vault"].pseudo_tag
-        }
+      image = {
+        repository = data.oci_string.ref["vault-k8s"].registry_repo
+        tag        = data.oci_string.ref["vault-k8s"].pseudo_tag
       }
-    })
-  ]
-}
-
-data "oci_exec_test" "acceptance" {
-  digest     = var.digests["vault"]
-  script     = "${path.module}/acceptance.sh"
-  depends_on = [helm_release.vault]
-  env {
-    name  = "NAME"
-    value = "vault-${random_pet.suffix.id}"
-  }
-  env {
-    name  = "NAMESPACE"
-    value = "vault-system-${random_pet.suffix.id}"
+    }
+    server = {
+      image = {
+        repository = data.oci_string.ref["vault"].registry_repo
+        tag        = data.oci_string.ref["vault"].pseudo_tag
+      }
+    }
   }
 }
 
-module "helm_cleanup" {
-  depends_on = [data.oci_exec_test.acceptance]
-  source     = "../../../tflib/helm-cleanup"
-  name       = helm_release.vault.id
-  namespace  = helm_release.vault.namespace
+resource "imagetest_feature" "basic" {
+  harness     = imagetest_harness_k3s.this
+  name        = "Basic"
+  description = "Basic functionality of the vault image."
+
+  steps = [
+    {
+      name = "Helm install vault"
+      cmd  = module.helm.install_cmd
+    },
+    {
+      name = "Acceptance test"
+      cmd  = <<EOF
+apk add jq
+
+KEY=$(kubectl exec -n vault-system vault-0 -- vault operator init \
+  -key-shares=1 \
+  -key-threshold=1 \
+  -format=json | jq -r ".unseal_keys_b64[]")
```

Comment:
Without explicitly stating 'bash' and 'set -o pipefail' this is just going to swallow errors.
I really just prefer doing this sort of thing in 2 steps.

```
out=$(kubectl exec -n vault-system vault-0 -- vault operator init \
  -key-shares=1 \
  -key-threshold=1 \
  -format=json) || { echo "kubectl exec vault failed: $out"; exit 1; }
 key=$(echo "$out" | jq -r ".unseal_keys_b64[]") || { echo "json parsing failed. doc=$out"; exit 1; }
  ```

Reply:
and what is the harm? invalid key? then unseal will fail, and test fails.

Reply:
and the error message is hidden down below the thing that actually went wrong.